internal/manifest/parse.go

```diff
@@ -115,3 +146,60 @@ func pullLayer(ctx context.Context, imageRef string, keyChain authn.Keychain) (c
 func getFsChartPath(imageSpec v1beta2.ImageSpec) string {
 	return filepath.Join(os.TempDir(), fmt.Sprintf("%s-%s", imageSpec.Name, imageSpec.Ref))
 }
+
+func untarLayer(tarPath string) (string, error) {
```

Comment:
This logic should be improved, you need to consider concurrency here, it will be multiple threads trying to read and uncompress the file, we configured multiple Manifest workers.

Please reference how the `getPathForFetchedLayer` is implemented, use `filemutex.MutexCache` could help in your case as well.

Reply:
Also, please write a test for this function, what's expected for this test:
1. use your `rendered.tar` as input,
2. initailze several goruntine, execute this untarLayer function.
3. verify the file:
   1.  only created once
   2. the completeness of the content in the file
   3. no modification after created 
 

I think you can use `FileInfo.ModTime` to verify 3.iii, it's returned by `io.OpenFile`
```
type FileInfo interface {
	Name() string       // base name of the file
	Size() int64        // length in bytes for regular files; system-dependent for others
	Mode() FileMode     // file mode bits
	ModTime() time.Time // modification time
	IsDir() bool        // abbreviation for Mode().IsDir()
	Sys() any           // underlying data source (can return nil)
}
```